cypress/integration/10_workflow_transitions.spec.js

```diff
@@ -0,0 +1,163 @@
+import * as dkan from '../support/helpers/dkan'
+
+context('Draft datasets', () => {
+  beforeEach(() => cy.drupalLogin('testeditor', 'testeditor'))
+
+  it('Draft datasets are hidden from the catalog until published.', () => {
+    // Create draft dataset
+    const dataset_title = dkan.generateRandomString()
+    dkan.createDatasetWithModerationState(dataset_title, 'draft')
+    cy.get('@datasetId').then(datasetId => {
+
+      // Ensure dataset is hidden from catalog
+      dkan.searchMetastore({fulltext: dataset_title, facets: ''}).then((response) => {
+        expect(response.status).to.eq(200)
+        expect(response.body.results).to.be.empty
+      })
+
+      // Ensure dataset details are not available when directly visited
+      cy.request({
+        url: '/api/1/metastore/schemas/dataset/items/' + datasetId,
```

Comment:
Using `?show-reference-ids` shouldn't change how the metastore calls work and possibly show something, correct? I use this param in the frontend on most of my calls.

Reply:
No, that parameter is checked and acted on after the item has already been retrieved from the metastore; it's outside of the logic being tested here.